keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1361 @@
+   //
+   // +required
+   Minimum resource.Quantity `json:"minimum,omitempty" protobuf:"bytes,1,name=minimum"`
+
+
+   // Maximum specifies the maximum capacity that can be requested in a consumption request.
+   //
+   // +optional
+   Maximum *resource.Quantity `json:"maximum,omitempty" protobuf:"bytes,2,opt,name=maximum"`
+
+
+   // Step defines the incremental block size by which capacity can increase from the minimum.
+   //
+   // +optional
+   Step *resource.Quantity `json:"step,omitempty" protobuf:"bytes,3,opt,name=step"`
+}
+
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // Capacities define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   Capacities *CapacityRequirements `json:"capacities,omitempty" protobuf:"bytes,9,opt,name=capacities"`
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+   // Requests specifies the requested quantity of device capacities for a device request.
+   // +optional
+   Requests map[QualifiedName]resource.Quantity `json:"requests,omitempty" protobuf:"bytes,1,rep,name=requests"`
+
+
+   // Potentially enhancement field.
+   // Limits define the maximum amount of per-device resources allowed.
+   // This enables burstable usage when applicable.
+}
+
+
+// DeviceConstraint must have exactly one field set besides Requests.
+type DeviceConstraint struct {
+	...
+   // DistinctAttribute requires that all devices in question have this
+   // attribute and that its type and value are unique across those devices.
+   //
+   // For example, specify attribute name to get virtual devices from distinct shared physical devices.
+   //
+   // +optional
+   // +oneOf=ConstraintType
+   DistinctAttribute *FullyQualifiedName `json:"distinctAttribute,omitempty" protobuf:"bytes,3,opt,name=distinctAttribute"`
+}
+
+```
+
+#### ResourceClaimStatus's DeviceRequestAllocationResult
+
+```go
+
+type DeviceRequestAllocationResult struct {
+ ...
+
+
+   // Shared indicates whether the allocated device can be shared by multiple claims.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   Shared *bool `json:"shared,omitempty" protobuf:"bytes,7,opt,name=shared"`
+
+
+   // ConsumedCapacities is used for tracking the capacity consumed per device by the claim request.
+   // The total consumed capacity for each device must not exceed its corresponding available capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   ConsumedCapacities map[QualifiedName]resource.Quantity `json:"consumedCapacities,omitempty" protobuf:"bytes,8,rep,name=consumedCapacities"`
+}
+
+```
+
+### Scheduling enhancement
+- define [share-related types and functions](https://github.com/sunya-ch/kubernetes/blob/kep-5075/staging/src/k8s.io/dynamic-resource-allocation/structured/share.go).
```

Comment:
I think this entire section is too detailed. We don't want to do a code review as part of the KEP review.

Can you try to summarize in plain English?

Reply:
Thank you. I removed the code and replaced with summarized sentences.